### ISSUE TITLE
[comma-dangle] Difference between function declaration and function call?

### ISSUE THREAD

**Issue by tleunen** [eslint/eslint#7588]
Just a suggestion, if it would make sense to have a difference for the function definition and function call?
I'm asking that because sometimes it feels weird to have the comma. Especially in function which takes an arrow function.
```
something.map(url =>
  sthelse
);
```
With the rule enabled with multiline, it asks a comma after `sthelse`. Would it be possible to have an option to not do it? Would it make sense?

This is particularly weird when playing with jsx.
```
const component = renderer.create(
  <ESLintItem>
    <a href="http://eslint.org">ESLint</a>
  </ESLintItem>,
);
```

**Comment by ljharb** [comment]
The following won't complain about a trailing comma:
```js
something.map(url => (
  sthelse
));
```

**Comment by eslintbot** [comment]
Hi @tleunen, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by tleunen** [comment]
@vitorbal :/ What info do you need?

**Comment by not-an-aardvark** [comment]
@tleunen See [eslintbot's comment](https://github.com/eslint/eslint/issues/7588#issuecomment-260888163) above.

Sorry if this seems tedious. We get a lot of bug reports and feature requests, and it's much easier for us to triage them if they're in a consistent format with all of this information.

**Comment by johntran** [comment]
I can attest to this:

eslint: v3.10.2
eslint config: https://gist.github.com/johntran/3c02e9effdeedfde7896403dcb8fa150

Source code:
```
const arr = [1, 2, 3];
const addOne = arr.map(
  item => item + 1
);
```

```
file.js
  68:7   error  'addOne' is assigned a value but never used  no-unused-vars
  69:19  error  Missing trailing comma                       comma-dangle

✖ 2 problems (2 errors, 0 warnings)
```

Expected to not throw comma dangle error. This isn't valid JS. See: https://repl.it/E4Fo

**Comment by cmalard** [comment]
Having the same issue with :
```js
console.log(
  'string'
);
```

**Comment by ljharb** [comment]
@johntran the airbnb config requires trailing commas on function arguments.

Indeed, your arrow function, and @cmalard, your console.log argument, need trailing commas. It is indeed valid JS, it's just ES2017. The airbnb config expects you're using https://npmjs.com/babel-preset-airbnb

**Comment by cmalard** [comment]
(got it right after my post, thanks :-) )

**Comment by msmichellegar** [comment]
I was just reporting the same issue as you commented @ljharb. Thanks!

**Comment by johntran** [comment]
@ljharb Ah I see :). Should we close this then?

**Comment by ljharb** [comment]
No, @tleunen's original post is not incorrectly complaining about comma-dangle, it's looking for an exception for `comma-dangle` applying to functions with jsx, and with multiline implicit return arrow functions.

**Comment by nzakas** [comment]
Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
